Broken CI

The CI is currently broken:

```
  /opt/hostedtoolcache/Ruby/3.1.3/x64/bin/bundle config --local path /home/runner/work/kurren/kurren/vendor/bundle
  /opt/hostedtoolcache/Ruby/3.1.3/x64/bin/bundle config --local deployment true
  Cache key: setup-ruby-bundler-cache-v4-ubuntu-22.04-ruby-3.1.3-Gemfile.lock-ae6de06502ebfa5d1b969b415064be83ca3adf57bb87695625817aa0073c4189
  /opt/hostedtoolcache/Ruby/3.1.3/x64/bin/bundle install --jobs 4
  Installing amq-protocol 2.3.2
  Installing ast 2.4.2
  Using bundler 2.3.7
  Installing rbtree 0.4.6 with native extensions
  Installing set 1.0.3
  Installing coderay 1.1.3
  Installing diff-lcs 1.5.0
  Installing dotenv 2.8.1
  Installing faraday-net_http 3.0.2
  Using ruby2_keywords 0.0.5
  Installing json 2.6.2 with native extensions
  Installing mini_mime 1.1.1
  Installing method_source 1.0.0
  Installing timeout 0.3.0
  Installing parallel 1.22.1
  Installing pkg-config 1.4.7
  Installing rainbow 3.1.1
  Installing regexp_parser 2.6.1
  Installing rexml 3.2.5
  Installing rspec-support 3.12.0
  Installing ruby-progressbar 1.11.0
  Installing unicode-display_width 2.3.0
  Installing parser 3.1.2.1
  Installing faraday 2.7.1
  Installing mail 2.7.1
  Installing net-protocol 0.1.3
  Installing pry 0.14.1
  Installing xmlhash 1.3.8 with native extensions
  Installing rspec-core 3.12.0
  Installing rspec-expectations 3.12.0
  Installing rspec-mocks 3.12.0
  Installing rubocop-ast 1.23.0
  Installing net-smtp 0.3.3
  Installing rspec 3.12.0
  Installing sorted_set 1.0.3
  Installing bunny 2.20.1
  Installing rubocop 1.39.0
  Installing rubocop-performance 1.15.1
  Installing rubocop-rspec 2.15.0
  Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
  
  current directory:
  /home/runner/work/kurren/kurren/vendor/bundle/ruby/3.1.0/gems/xmlhash-1.3.8/ext/xmlhash
  /opt/hostedtoolcache/Ruby/3.1.3/x64/bin/ruby -I
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/3.1.0 extconf.rb
  checking for xmlAddID() in -lxml2... yes
  creating Makefile
  
  current directory:
  /home/runner/work/kurren/kurren/vendor/bundle/ruby/3.1.0/gems/xmlhash-1.3.8/ext/xmlhash
  make DESTDIR\= sitearchdir\=./.gem.20221220-1725-6dv8qz
  sitelibdir\=./.gem.20221220-1725-6dv8qz clean
  
  current directory:
  /home/runner/work/kurren/kurren/vendor/bundle/ruby/3.1.0/gems/xmlhash-1.3.8/ext/xmlhash
  make DESTDIR\= sitearchdir\=./.gem.20221220-1725-6dv8qz
  sitelibdir\=./.gem.20221220-1725-6dv8qz
  compiling xmlhash.c
  xmlhash.c: In function ‘init_XmlhashParserData’:
  xmlhash.c:29:6: error: old-style function definition
  [-Werror=old-style-definition]
     29 | void init_XmlhashParserData()
        |      ^~~~~~~~~~~~~~~~~~~~~~
  xmlhash.c: In function ‘Init_xmlhash’:
  xmlhash.c:241:6: error: old-style function definition
  [-Werror=old-style-definition]
    241 | void Init_xmlhash()
        |      ^~~~~~~~~~~~
  xmlhash.c: At top level:
  cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
  intended to silence earlier diagnostics
  cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
  been intended to silence earlier diagnostics
  cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
  have been intended to silence earlier diagnostics
  cc1: all warnings being treated as errors
  make: *** [Makefile:247: xmlhash.o] Error 1
  
  make failed, exit code 2
  
  Gem files will remain installed in
  /home/runner/work/kurren/kurren/vendor/bundle/ruby/3.1.0/gems/xmlhash-1.3.8 for
  inspection.
  Results logged to
  /home/runner/work/kurren/kurren/vendor/bundle/ruby/3.1.0/extensions/x86_64-linux/3.1.0/xmlhash-1.3.8/gem_make.out
  
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/3.1.0/rubygems/ext/builder.rb:102:in
  `run'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/3.1.0/rubygems/ext/builder.rb:51:in
  `block in make'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/3.1.0/rubygems/ext/builder.rb:43:in
  `each'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/3.1.0/rubygems/ext/builder.rb:43:in
  `make'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/3.1.0/rubygems/ext/ext_conf_builder.rb:42:in
  `build'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/3.1.0/rubygems/ext/builder.rb:171:in
  `build_extension'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/3.1.0/rubygems/ext/builder.rb:205:in
  `block in build_extensions'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/3.1.0/rubygems/ext/builder.rb:202:in
  `each'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/3.1.0/rubygems/ext/builder.rb:202:in
  `build_extensions'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/3.1.0/rubygems/installer.rb:843:in
  `build_extensions'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/rubygems_gem_installer.rb:71:in
  `build_extensions'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/rubygems_gem_installer.rb:28:in
  `install'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/source/rubygems.rb:204:in
  `install'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/installer/gem_installer.rb:54:in
  `install'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/installer/gem_installer.rb:16:in
  `install_from_spec'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/installer/parallel_installer.rb:186:in
  `do_install'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/installer/parallel_installer.rb:177:in
  `block in worker_pool'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/worker.rb:62:in
  `apply_func'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/worker.rb:57:in
  `block in process_queue'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/worker.rb:54:in
  `loop'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/worker.rb:54:in
  `process_queue'
  /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/lib/bundler/worker.rb:91:in
  `block (2 levels) in create_threads'
  
  An error occurred while installing xmlhash (1.3.8), and Bundler cannot continue.
  
  In Gemfile:
    xmlhash
  Took   5.48 seconds
Error: Error: The process '/opt/hostedtoolcache/Ruby/3.1.3/x64/bin/bundle' failed with exit code 5
    at ExecState._setResult (/home/runner/work/_actions/ruby/setup-ruby/v1/dist/index.js:5340:25)
    at ExecState.CheckComplete (/home/runner/work/_actions/ruby/setup-ruby/v1/dist/index.js:5323:18)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/ruby/setup-ruby/v1/dist/index.js:5217:27)
    at ChildProcess.emit (node:events:390:28)
    at maybeClose (node:internal/child_process:1064:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
```

--- a/.github/workflows/rubocop.yml
+++ b/.github/workflows/rubocop.yml
@@ -13,6 +13,6 @@ jobs:
       - uses: actions/checkout@v2
       - uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 3.1
+          ruby-version: 3.1.2
           bundler-cache: true
       - run: bundle exec rubocop

--- a/.github/workflows/spec.yml
+++ b/.github/workflows/spec.yml
@@ -15,7 +15,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 3.1
+          ruby-version: 3.1.2
           bundler-cache: true
       - name: Run tests
         run: |

